### PR TITLE
feat: track http request in az func http trigger

### LINKED ISSUE #612: Assign correlation information to telemetry context in Azure Functions HTTP trigger template
**Is your feature request related to a problem? Please describe.**
Currently, the correlation enricher is not added to Azure Functions projects as we are missing the `IServiceProvider`, but we do have access to the telemetry context, we can help with adding correlation to the requests (and so support service-to-service correlation).

**Describe the solution you'd like**
Assign the correlation information to the telemetry context (use the default `OperationId` and `TransactionId` as these are defaults) and make sure that the context is passed along to every telemetry tracking.

```csharp
telemetryContext.Add("TransactionId", correlationInfo.TransactionId);
telemetryContext.Add("OperationId", correlationInfo.OperationId);
```

Discovered by my fellow partner @pim-simons! 🎊

### PR DESCRIPTION
**Pull request by stijnmoreels**
Tracks the incoming HTTP request in the Azure Functions HTTP trigger project template. It uses a simple trick to enable service-to-service correlation by enriching the telemetry context with the correlation information instead of a dedicated Serilog enricher.

Closes #612

### CODE REVIEW

**Comment by stijnmoreels** [comment]
Won't probably be able to do it via the Serilog `LogContext` as the correlation happens later than the request tracking, so the way it is setup is probably the best way.
Thx @pim-simons for investigating this! 👍

**Review by pim-simons — APPROVED** [review]
Looks good to me 🥇

File: src/Arcus.Templates.AzureFunctions.Http/OrderFunction.cs [hunk 1]
```diff
@@ -52,39 +54,55 @@ public OrderFunction(ISecretProvider secretProvider, HttpCorrelation httpCorrela
...
+                    Logger.LogInformation("C# HTTP trigger 'order' function processed a request");
+                    if (IsJson(request) == false || AcceptsJson(request) == false)
+                    {
+                        string accept = string.Join(", ", GetAcceptMediaTypes(request));
+                        Logger.LogError("Could not process current request because the request body is not JSON and/or could not accept JSON as response (Content-Type: {ContentType}, Accept: {Accept})", request.ContentType, accept);
 
-                    return UnsupportedMediaType("Could not process current request because the request body is not JSON and/or could not accept JSON as response");
-                }
+                        statusCode = StatusCodes.Status415UnsupportedMediaType;
+                        return Error(statusCode, "Could not process current request because the request body is not JSON and/or could not accept JSON as response");
```
[the remark this answers is not in the document]
Comment (stijnmoreels) [reply src/Arcus.Templates.AzureFunctions.Http/OrderFunction.cs:71]: Two things are checked here: 
1. if the received request body is JSON (`Content-Type`, and
2. if the Azure Function can respond with JSON (`Accept`).

[the remark this answers is not in the document]
Comment (fgheysels) [reply src/Arcus.Templates.AzureFunctions.Http/OrderFunction.cs:71]: I don't remember anymore what I meant with this comment

**Review by fgheysels — APPROVED** [review]

**Review by fgheysels — APPROVED** [review]

**Pull request merged**